packages/rspack/src/plugins/utils/apply-base-config.ts

```diff
@@ -240,8 +240,11 @@ function applyNxDependentConfig(
     plugins.push(new NxTsconfigPathsRspackPlugin({ ...options, tsConfig }));
   }
 
-  // New TS Solution already has a typecheck target
-  if (!options?.skipTypeChecking && !isUsingTsSolution) {
+  // New TS Solution already has a typecheck target but allow it to run during serve
+  if (
+    !options?.skipTypeChecking &&
+    (!isUsingTsSolution || process.env['WEBPACK_SERVE'])
+  ) {
```

Comment:
The default value for `skipTypeChecking` is `false` but for the TS solution setup we should consider it to be `true`. Let's explicitly check to `skipTypeChecking === false`:
```suggestion
  if (
    (!options?.skipTypeChecking && !isUsingTsSolution) ||
    (isUsingTsSolution &&
      options?.skipTypeChecking === false &&
      process.env['WEBPACK_SERVE'])
  ) {
```